core/src/main/java/org/opentosca/toscana/core/transformation/Property.java

```diff
@@ -0,0 +1,54 @@
+package org.opentosca.toscana.core.transformation;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+public class Property<T> {
```

Comment:
i would not use a generic type for Properties! This makes it impossible to differentiate passwords from regular strings.

I think the following solution could be good.
Instead of making Properties generic i would just store the values in Strings. Typing them is no good idea especially if we want to put them in a Properties file, which basically is a String to String Map.
The type would be stored in a PropertyType Enum containing the following types:
* ``TEXT`` - Just a regular string
* ``IDENTIFIER`` -  a Short string only containing lower case letters, numbers (not at the beginning) and dashes
* ``PASSWORD`` - A String that is "confidential" and should not be shown in cleartext on screen. This mostly is a frontend indicator to tell the CLI or GUI to hide the input with `*`
* ``URL`` -  A url that will point to something
* ``NUMBER`` - A Integer Number (Internal Datatype will probably be Long)
* ``FLOAT`` - Floating Point Number (Internal Datatype: Double)

The propery class will contain the following attributes (when following my approach):
```Java
//This gets set to true if the property is required for transformation
//However replacing it with a enum to differentiate between a Required Property to run the Transformation or a Deployment Property
private boolean required; 
//The type of the Property
private PropertyType type;
//The identifying key of the property
private String key;
```
This approach also seperates the schema of the properties and their values Storing the values can be aly easy as putting them into a seperate Map<Property, String> this could also be wrpapped somehow tough.